File: movaro/apps/api/src/modules/city-insights/domain/entities/city-insight.entity.ts
export enum CityInsightType {
  Lifestyle = 'lifestyle',
  CostOfLiving = 'cost_of_living',
  Housing = 'housing',
  Work = 'work',
  PracticalTip = 'practical_tip',
  Motivation = 'motivation',
}

export enum CityInsightTheme {
  Neighborhoods = 'neighborhoods',
  Beaches = 'beaches',
  Parks = 'parks',
  Nightlife = 'nightlife',
  FoodAndCafes = 'food_and_cafes',
  CultureAndEvents = 'culture_and_events',
  Viewpoints = 'viewpoints',
  LocalRoutine = 'local_routine',
}

export type CityInsightSource = 'curated' | 'generated' | 'template';

export interface CityInsightEntity {
  id: string;
  cityId: string;
  cityName: string;
  type: CityInsightType;
  theme?: CityInsightTheme;
  title: string;
  shortText: string;
  content: string;
  placeHighlights?: string[];
  imageUrl?: string;
  ctaLabel?: string;
  source: CityInsightSource;
  generatedAt: string;
  expiresAt?: string;
}

File: movaro/apps/app/lib/features/city_insights/domain/entities/city_insight_entity.dart
enum CityInsightType {
  lifestyle,
  costOfLiving,
  housing,
  work,
  practicalTip,
  motivation,
}

enum CityInsightTheme {
  neighborhoods,
  beaches,
  parks,
  nightlife,
  foodAndCafes,
  cultureAndEvents,
  viewpoints,
  localRoutine,
}

class CityInsightEntity {
  const CityInsightEntity({
    required this.id,
    required this.cityId,
    required this.cityName,
    required this.type,
    required this.title,
    required this.shortText,
    required this.content,
    required this.source,
    required this.generatedAt,
    this.theme,
    this.placeHighlights = const [],
    this.imageUrl,
    this.ctaLabel,
    this.expiresAt,
  });

  final String id;
  final String cityId;
  final String cityName;
  final CityInsightType type;
  final CityInsightTheme? theme;
  final String title;
  final String shortText;
  final String content;
  final String source;
  final DateTime generatedAt;
  final List<String> placeHighlights;
  final String? imageUrl;
  final String? ctaLabel;
  final DateTime? expiresAt;
}
